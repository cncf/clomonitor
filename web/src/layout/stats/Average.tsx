import { CATEGORY_ICONS } from '../../data';
import { ScoreType } from '../../types';
import styles from './Average.module.css';
import ProgressBar from './ProgressBar';

interface Props {
  title: string;
  data: { [key in ScoreType]: number };
}

const Average = (props: Props) => {
  return (
    <>
      <div className={`card-header fw-bold text-uppercase text-center ${styles.cardHeader}`}>{props.title}</div>
      <div className="card-body pt-3 px-4 pb-0">
        <ProgressBar
          title="Documentation"
          icon={CATEGORY_ICONS[ScoreType.Documentation]}
          value={props.data.documentation}
        />
        <ProgressBar title="License" icon={CATEGORY_ICONS[ScoreType.License]} value={props.data.license} />
        <ProgressBar
          title="Best Practices"
          icon={CATEGORY_ICONS[ScoreType.BestPractices]}
          value={props.data.best_practices}
        />
        <ProgressBar title="Security" icon={CATEGORY_ICONS[ScoreType.Security]} value={props.data.security} />
        <ProgressBar title="Legal" icon={CATEGORY_ICONS[ScoreType.Legal]} value={props.data.legal} />
      </div>
    </>
  );
};

export default Average;
