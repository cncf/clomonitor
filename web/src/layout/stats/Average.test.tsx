import { render, screen } from '@testing-library/react';

import Average from './Average';

const defaultProps = {
  data: { documentation: 85, license: 79, best_practices: 57, security: 53, legal: 40, global: 69 },
  title: 'Sandbox',
};

describe('Average', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Average {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      render(<Average {...defaultProps} />);

      expect(screen.getByText('Sandbox')).toBeInTheDocument();
      expect(screen.getByText('Documentation')).toBeInTheDocument();
      expect(screen.getByText('85%')).toBeInTheDocument();
      expect(screen.getByText('License')).toBeInTheDocument();
      expect(screen.getByText('79%')).toBeInTheDocument();
      expect(screen.getByText('Best Practices')).toBeInTheDocument();
      expect(screen.getByText('57%')).toBeInTheDocument();
      expect(screen.getByText('Security')).toBeInTheDocument();
      expect(screen.getByText('53%')).toBeInTheDocument();
      expect(screen.getByText('Legal')).toBeInTheDocument();
      expect(screen.getByText('40%')).toBeInTheDocument();

      expect(screen.getAllByRole('progressbar')).toHaveLength(5);
    });
  });
});
