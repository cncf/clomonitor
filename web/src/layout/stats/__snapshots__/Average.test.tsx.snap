// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="card-header fw-bold text-uppercase text-center cardHeader"
  >
    Sandbox
  </div>
  <div
    class="card-body pt-2 pt-md-3 px-3 px-md-4 pb-0"
  >
    <div
      class="d-flex flex-column"
    >
      <div
        class="d-flex flex-row align-items-center mb-1 mb-md-0 progressTitle"
      >
        <div
          class="me-2 position-relative icon"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <div>
          Documentation
        </div>
      </div>
      <div
        class="d-flex flex-row mb-3 align-items-center"
      >
        <div
          class="flex-grow-1 progressbarWrapper"
        >
          <div
            class="progress rounded-0 progress"
          >
            <div
              class="progress-bar"
              role="progressbar"
              style="width: 85%;"
            />
          </div>
        </div>
        <div
          class="ps-1 ps-md-3 lh-1 text-end scoreWrapper"
        >
          <small
            class="fw-bold lightText"
          >
            85%
          </small>
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <div
        class="d-flex flex-row align-items-center mb-1 mb-md-0 progressTitle"
      >
        <div
          class="me-2 position-relative icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32c-70.7 0-128 57.3-128 128s57.3 128 128 128 128-57.3 128-128S326.7 32 256 32zm0 208c-44.2 0-80-35.8-80-80s35.8-80 80-80 80 35.8 80 80-35.8 80-80 80zM193.7 307.4c-19.1-8.1-36.2-19.6-50.8-34.3-1.4-1.4-2.8-2.8-4.1-4.3L64 400h96l48 80 48-105.8 25.5-56.2c-8.4 1.3-16.9 2-25.5 2-21.6 0-42.5-4.2-62.3-12.6zM373.3 268.9c-1.3 1.4-2.7 2.9-4.1 4.3-14.6 14.6-31.7 26.2-50.7 34.2L294 361.2l-21.9 48.4L304 480l48-80h96l-74.7-131.1z"
            />
          </svg>
        </div>
        <div>
          License
        </div>
      </div>
      <div
        class="d-flex flex-row mb-3 align-items-center"
      >
        <div
          class="flex-grow-1 progressbarWrapper"
        >
          <div
            class="progress rounded-0 progress"
          >
            <div
              class="progress-bar"
              role="progressbar"
              style="width: 79%;"
            />
          </div>
        </div>
        <div
          class="ps-1 ps-md-3 lh-1 text-end scoreWrapper"
        >
          <small
            class="fw-bold lightText"
          >
            79%
          </small>
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <div
        class="d-flex flex-row align-items-center mb-1 mb-md-0 progressTitle"
      >
        <div
          class="me-2 position-relative icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5 4.604v9.185a4 4 0 0 0 1.781 3.328L12 20.597l5.219-3.48A4 4 0 0 0 19 13.79V4.604L12 3.05 5 4.604zM3.783 2.826L12 1l8.217 1.826a1 1 0 0 1 .783.976v9.987a6 6 0 0 1-2.672 4.992L12 23l-6.328-4.219A6 6 0 0 1 3 13.79V3.802a1 1 0 0 1 .783-.976zM12 13.5l-2.939 1.545.561-3.272-2.377-2.318 3.286-.478L12 6l1.47 2.977 3.285.478-2.377 2.318.56 3.272L12 13.5z"
              />
            </g>
          </svg>
        </div>
        <div>
          Best Practices
        </div>
      </div>
      <div
        class="d-flex flex-row mb-3 align-items-center"
      >
        <div
          class="flex-grow-1 progressbarWrapper"
        >
          <div
            class="progress rounded-0 progress"
          >
            <div
              class="progress-bar"
              role="progressbar"
              style="width: 57%;"
            />
          </div>
        </div>
        <div
          class="ps-1 ps-md-3 lh-1 text-end scoreWrapper"
        >
          <small
            class="fw-bold lightText"
          >
            57%
          </small>
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <div
        class="d-flex flex-row align-items-center mb-1 mb-md-0 progressTitle"
      >
        <div
          class="me-2 position-relative icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C9.243 2 7 4.243 7 7v2H6c-1.103 0-2 .897-2 2v9c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-9c0-1.103-.897-2-2-2h-1V7c0-2.757-2.243-5-5-5zM9 7c0-1.654 1.346-3 3-3s3 1.346 3 3v2H9V7zm9.002 13H13v-2.278c.595-.347 1-.985 1-1.722 0-1.103-.897-2-2-2s-2 .897-2 2c0 .736.405 1.375 1 1.722V20H6v-9h12l.002 9z"
            />
          </svg>
        </div>
        <div>
          Security
        </div>
      </div>
      <div
        class="d-flex flex-row mb-3 align-items-center"
      >
        <div
          class="flex-grow-1 progressbarWrapper"
        >
          <div
            class="progress rounded-0 progress"
          >
            <div
              class="progress-bar"
              role="progressbar"
              style="width: 53%;"
            />
          </div>
        </div>
        <div
          class="ps-1 ps-md-3 lh-1 text-end scoreWrapper"
        >
          <small
            class="fw-bold lightText"
          >
            53%
          </small>
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <div
        class="d-flex flex-row align-items-center mb-1 mb-md-0 progressTitle"
      >
        <div
          class="me-2 position-relative icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4c-.83 0-1.5-.67-1.5-1.5S6.17 1 7 1s1.5.67 1.5 1.5S7.83 4 7 4zm7 6c0 1.11-.89 2-2 2h-1c-1.11 0-2-.89-2-2l2-4h-1c-.55 0-1-.45-1-1H8v8c.42 0 1 .45 1 1h1c.42 0 1 .45 1 1H3c0-.55.58-1 1-1h1c0-.55.58-1 1-1h.03L6 5H5c0 .55-.45 1-1 1H3l2 4c0 1.11-.89 2-2 2H2c-1.11 0-2-.89-2-2l2-4H1V5h3c0-.55.45-1 1-1h4c.55 0 1 .45 1 1h3v1h-1l2 4zM2.5 7L1 10h3L2.5 7zM13 10l-1.5-3-1.5 3h3z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div>
          Legal
        </div>
      </div>
      <div
        class="d-flex flex-row mb-3 align-items-center"
      >
        <div
          class="flex-grow-1 progressbarWrapper"
        >
          <div
            class="progress rounded-0 progress"
          >
            <div
              class="progress-bar"
              role="progressbar"
              style="width: 40%;"
            />
          </div>
        </div>
        <div
          class="ps-1 ps-md-3 lh-1 text-end scoreWrapper"
        >
          <small
            class="fw-bold lightText"
          >
            40%
          </small>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
